src/harness/fakesHosts.ts

```diff
@@ -532,7 +532,7 @@ ${indentText}${text}`;
 
         private constructor(sys: System | vfs.FileSystem, options?: ts.CompilerOptions, setParentNodes?: boolean, createProgram?: ts.CreateProgram<ts.BuilderProgram>) {
             super(sys, options, setParentNodes);
-            this.createProgram = createProgram || ts.createEmitAndSemanticDiagnosticsBuilderProgram;
+            this.createProgram = createProgram || ts.createEmitAndSemanticDiagnosticsBuilderProgram as unknown as ts.CreateProgram<ts.BuilderProgram>;

```

Comment:
Why the up-cast?

Reply:
It mirrors the one here: https://github.com/microsoft/TypeScript/blob/5a53e9bb5ecb96f79cd55f7a4fb632d235582eb0/src/compiler/watch.ts#L661

But, I do think this is weird. I'll check to see if there's a way to avoid all of these casts.

Reply:
I'm not sure how to avoid this without breaking the public API here quite a bit; I'm not super comfortable working with the Program, Builder, etc, types, and there really a lot of casting going on that I can't reason about too well. I think that things could be better, but I think fixing that is going to be non-trivial.